src/deluge/model/sample/sample_recorder.cpp

```diff
@@ -696,8 +696,8 @@ Error SampleRecorder::finalizeRecordedFile() {
 				if (firstSampleCluster->sdAddress == 0) {
 					FREEZE_WITH_ERROR("E268");
 				}
-				if ((firstSampleCluster->sdAddress - fileSystemStuff.fileSystem->database)
-				    & (fileSystemStuff.fileSystem->csize - 1)) {
+				if ((firstSampleCluster->sdAddress - fileSystemStuff.fileSystem.value().database)
```

Comment:
I don't think calling .value() is safe either - https://en.cppreference.com/w/cpp/utility/optional/value

Docs say it throws if the optional isn't present

Reply:
This isn't UB (unlike ->) but will crash deluge

Reply:
if the filesystem isn't valid I'd rather it crash than try to corrupt the SDCard or something.